lib/debloater/index.rb

```diff
@@ -0,0 +1,100 @@
+require 'debloater/helpers'
+
+module Debloater
+  class Index
+    include Helpers
+
+    # conn: connection object
+    # method: the function used to obtain index information
+    # name: the index name
+    def initialize(conn, name:)
+      @conn = conn
+      @name = name
+    end
+
+    attr_reader :name
+
+    def self.each(conn)
+      data = conn.exec_params %{
+        select * from pg_indexes where schemaname = $1
+      }, ['public']
+      data.sort_by { |d| 
+        d['indexname'] 
+      }.each { |d| 
+        yield new(conn, name: d['indexname'])
+      }
+    rescue PG::Error => e
+      _fatal e, msg: 'Could not list indexes'
+    end
+
+
+    def pkey?
+      @name =~ /pkey/
+    end
+
+
+    def valid?
+      !_metadata.nil?
+    end
+
+
+    def density
+      @density ||= 0.01 * _metadata['avg_leaf_density'].to_f
+    end
+
+
+    def bloat
+      @bloat ||= size * (1 - density)
+    end
+
+
+    def size
+      @size ||= _metadata['index_size'].to_i / 1024**2
+    end
+
+
+    def debloat!
+      [
+        index['indexdef'].
+          sub('CREATE INDEX', 'CREATE INDEX CONCURRENTLY').
```

Comment:
Is it possible that `index['indexdef']` already contains `CREATE INDEX CONCURRENTLY`? Is there any chance that this will result in a duplicated modifier, i.e. `CREATE INDEX CONCURRENTLY CONCURRENTLY`?